Quarkus REST client and Quarkus REST with abstract resource - NATIVE build fails

### Describe the bug

I'm having app with Quarkus REST client and Quarkus REST with abstract resources with `@Path` and the build of native binary fails

If Quarkus REST client is removed, native compilation works.

Details:
```
Build resources:
 - 26.49GB of memory (73.6% of 36.00GB system memory, determined at start)
 - 12 thread(s) (100.0% of 12 available processor(s), determined at start)
[2/8] Performing analysis...  [08:12:56,063 WARN  [io.net.res.dns.DnsServerAddressStreamProviders] Can not find io.netty.resolver.dns.macos.MacOSDnsServerAddressStreamProvider in the classpath, fallback to system defaults. This may result in incorrect DNS resolutions on MacOS. Check whether you have a dependency on 'io.netty:netty-resolver-dns-native-macos'
*****]                                                                   (10.5s @ 0.87GB)
   11,071 reachable types   (86.0% of   12,874 total)
   15,842 reachable fields  (56.1% of   28,250 total)
   55,945 reachable methods (56.2% of   99,630 total)
    3,616 types,   118 fields, and 3,364 methods registered for reflection
       62 types,    67 fields, and    55 methods registered for JNI access
        5 native libraries: -framework CoreServices, -framework Foundation, dl, pthread, z

Error: Discovered unresolved type during parsing: org.acme.AbstractGreetingResource$$QuarkusRestClientInterface. This error is reported at image build time because class org.acme.AbstractGreetingResource$$QuarkusRestClientInterfaceCreator is registered for linking at image build time by command line and command line.
Error encountered while parsing org.acme.AbstractGreetingResource$$QuarkusRestClientInterfaceCreator.apply(Unknown Source)
Parsing context:
   at io.quarkus.vertx.http.runtime.VertxHttpRecorder$18.accept(VertxHttpRecorder.java:1626)
   at io.quarkus.vertx.http.runtime.VertxHttpRecorder$18.accept(VertxHttpRecorder.java:1622)
   at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
   at io.smallrye.config.ConfigMappingContext$ObjectCreator.optionalGroup(ConfigMappingContext.java:641)
   at io.quarkus.tls.runtime.config.TlsBucketConfig351136998Impl.<init>(Unknown Source)
   at com.oracle.svm.core.code.FactoryMethodHolder.TlsBucketConfig351136998Impl_constructor_ee647b8c465636ec0f197409c5afe32ea7a63275(generated:0)
   at static root method.(Unknown Source)

Detailed message:

com.oracle.svm.core.util.UserError$UserException: Discovered unresolved type during parsing: org.acme.AbstractGreetingResource$$QuarkusRestClientInterface. This error is reported at image build time because class org.acme.AbstractGreetingResource$$QuarkusRestClientInterfaceCreator is registered for linking at image build time by command line and command line.
Error encountered while parsing org.acme.AbstractGreetingResource$$QuarkusRestClientInterfaceCreator.apply(Unknown Source)
Parsing context:
   at io.quarkus.vertx.http.runtime.VertxHttpRecorder$18.accept(VertxHttpRecorder.java:1626)
   at io.quarkus.vertx.http.runtime.VertxHttpRecorder$18.accept(VertxHttpRecorder.java:1622)
   at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
   at io.smallrye.config.ConfigMappingContext$ObjectCreator.optionalGroup(ConfigMappingContext.java:641)
   at io.quarkus.tls.runtime.config.TlsBucketConfig351136998Impl.<init>(Unknown Source)
   at com.oracle.svm.core.code.FactoryMethodHolder.TlsBucketConfig351136998Impl_constructor_ee647b8c465636ec0f197409c5afe32ea7a63275(generated:0)
   at static root method.(Unknown Source)

Detailed message:

	at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.UserError.abort(UserError.java:85)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FallbackFeature.reportAsFallback(FallbackFeature.java:248)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:814)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:592)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:550)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:539)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:721)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.start(NativeImageGeneratorRunner.java:143)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:98)
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Discovered unresolved type during parsing: org.acme.AbstractGreetingResource$$QuarkusRestClientInterface. This error is reported at image build time because class org.acme.AbstractGreetingResource$$QuarkusRestClientInterfaceCreator is registered for linking at image build time by command line and command line.
Error encountered while parsing org.acme.AbstractGreetingResource$$QuarkusRestClientInterfaceCreator.apply(Unknown Source)
Parsing context:
   at io.quarkus.vertx.http.runtime.VertxHttpRecorder$18.accept(VertxHttpRecorder.java:1626)
   at io.quarkus.vertx.http.runtime.VertxHttpRecorder$18.accept(VertxHttpRecorder.java:1622)
   at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
   at io.smallrye.config.ConfigMappingContext$ObjectCreator.optionalGroup(ConfigMappingContext.java:641)
   at io.quarkus.tls.runtime.config.TlsBucketConfig351136998Impl.<init>(Unknown Source)
   at com.oracle.svm.core.code.FactoryMethodHolder.TlsBucketConfig351136998Impl_constructor_ee647b8c465636ec0f197409c5afe32ea7a63275(generated:0)
   at static root method.(Unknown Source)

Detailed message:

	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.constraints.UnsupportedFeatures.report(UnsupportedFeatures.java:126)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:809)
	... 6 more
Caused by: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved type during parsing: org.acme.AbstractGreetingResource$$QuarkusRestClientInterface. This error is reported at image build time because class org.acme.AbstractGreetingResource$$QuarkusRestClientInterfaceCreator is registered for linking at image build time by command line and command line.
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.reportUnresolvedElement(SharedGraphBuilderPhase.java:550)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.reportUnresolvedElement(SharedGraphBuilderPhase.java:544)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.handleUnresolvedType(SharedGraphBuilderPhase.java:436)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.handleUnresolvedNewInstance(SharedGraphBuilderPhase.java:293)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genNewInstance(BytecodeParser.java:4641)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genNewInstance(BytecodeParser.java:4634)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBytecode(BytecodeParser.java:5444)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3431)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.iterateBytecodesForBlock(SharedGraphBuilderPhase.java:743)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.handleBytecodeBlock(BytecodeParser.java:3391)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBlock(BytecodeParser.java:3233)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.build(BytecodeParser.java:1137)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.build(SharedGraphBuilderPhase.java:162)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.buildRootMethod(BytecodeParser.java:1029)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.GraphBuilderPhase$Instance.run(GraphBuilderPhase.java:101)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase.run(SharedGraphBuilderPhase.java:116)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.run(Phase.java:49)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:434)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:42)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:38)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.AnalysisParsedGraph.parseBytecode(AnalysisParsedGraph.java:146)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.parseGraph(AnalysisMethod.java:895)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsedHelper(AnalysisMethod.java:860)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsed(AnalysisMethod.java:843)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.parse(MethodTypeFlowBuilder.java:186)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.apply(MethodTypeFlowBuilder.java:621)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.createFlowsGraph(MethodTypeFlow.java:167)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.ensureFlowsGraphCreated(MethodTypeFlow.java:153)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.getOrCreateMethodFlowsGraphInfo(MethodTypeFlow.java:111)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.typestate.DefaultVirtualInvokeTypeFlow.onObservedUpdate(DefaultVirtualInvokeTypeFlow.java:114)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.TypeFlow.update(TypeFlow.java:620)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.PointsToAnalysis$1.run(PointsToAnalysis.java:491)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:187)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:171)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1423)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
```

`AbstractGreetingResource$$QuarkusRestClientInterfaceCreator` looks a bit strange.

If I switch to quarkus-resteasy-client and quarkus-resteasy, things work well.

### Expected behavior

Native compilation works.

### Actual behavior

Native compilation fails

### How to Reproduce?

 - `sdk use java 21.0.2-graalce`
 - `git clone git@github.com:rsvoboda/abstract-quarkus-rest-repro.git`
 - `cd abstract-quarkus-rest-repro`
 - `mvn clean verify -Dnative`


### Output of `uname -a` or `ver`

macOS

### Output of `java -version`

Java 21

### Quarkus version or git rev

Quarkus main / Quarkus 3.14.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/scanning/ResteasyReactiveScanner.java
+++ b/independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/scanning/ResteasyReactiveScanner.java
@@ -324,7 +324,7 @@ public class ResteasyReactiveScanner {
                     MethodInfo annotatedMethod = methodAnnotationInstance.target().asMethod();
                     ClassInfo classWithJaxrsMethod = annotatedMethod.declaringClass();
                     if (Modifier.isAbstract(annotatedMethod.flags())
-                            && Modifier.isAbstract(classWithJaxrsMethod.flags())
+                            && Modifier.isInterface(classWithJaxrsMethod.flags())
                             && !clientInterfaces.containsKey(classWithJaxrsMethod.name())) {
                         clientInterfaces.put(classWithJaxrsMethod.name(), "");
                     }

--- a/independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/scanning/ResteasyReactiveScanner.java
+++ b/independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/scanning/ResteasyReactiveScanner.java
@@ -42,29 +42,20 @@ import org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames;
 
 public class ResteasyReactiveScanner {
 
-    public static final Map<DotName, String> BUILTIN_HTTP_ANNOTATIONS_TO_METHOD;
-    public static final Map<String, DotName> METHOD_TO_BUILTIN_HTTP_ANNOTATIONS;
-
-    static {
-        Map<DotName, String> map = new HashMap<>();
-        Map<String, DotName> reverseMap = new HashMap<>();
-        map.put(GET, "GET");
-        reverseMap.put("GET", GET);
-        map.put(POST, "POST");
-        reverseMap.put("POST", POST);
-        map.put(HEAD, "HEAD");
-        reverseMap.put("HEAD", HEAD);
-        map.put(PUT, "PUT");
-        reverseMap.put("PUT", PUT);
-        map.put(DELETE, "DELETE");
-        reverseMap.put("DELETE", DELETE);
-        map.put(PATCH, "PATCH");
-        reverseMap.put("PATCH", PATCH);
-        map.put(OPTIONS, "OPTIONS");
-        reverseMap.put("OPTIONS", OPTIONS);
-        BUILTIN_HTTP_ANNOTATIONS_TO_METHOD = Collections.unmodifiableMap(map);
-        METHOD_TO_BUILTIN_HTTP_ANNOTATIONS = Collections.unmodifiableMap(reverseMap);
-    }
+    public static final Map<DotName, String> BUILTIN_HTTP_ANNOTATIONS_TO_METHOD = Map.of(GET, "GET",
+            POST, "POST",
+            HEAD, "HEAD",
+            PUT, "PUT",
+            DELETE, "DELETE",
+            PATCH, "PATCH",
+            OPTIONS, "OPTIONS");
+    public static final Map<String, DotName> METHOD_TO_BUILTIN_HTTP_ANNOTATIONS = Map.of("GET", GET,
+            "POST", POST,
+            "HEAD", HEAD,
+            "PUT", PUT,
+            "DELETE", DELETE,
+            "PATCH", PATCH,
+            "OPTIONS", OPTIONS);
 
     public static ApplicationScanningResult scanForApplicationClass(IndexView index, Set<String> excludedClasses) {
         Collection<ClassInfo> applications = index
@@ -277,7 +268,7 @@ public class ResteasyReactiveScanner {
         });
 
         Map<DotName, String> clientInterfaces = new HashMap<>(pathInterfaces);
-        // for clients it is enough to have @PATH annotations on methods only
+        // for clients, it is enough to have @PATH annotations on methods only
         for (DotName interfaceName : interfacesWithPathOnMethods) {
             if (!clientInterfaces.containsKey(interfaceName)) {
                 clientInterfaces.put(interfaceName, "");
@@ -325,7 +316,7 @@ public class ResteasyReactiveScanner {
             httpAnnotationToMethod.put(httpMethodInstance.target().asClass().name(), httpMethodInstance.value().asString());
         }
 
-        // for clients it is also enough to only have @GET, @POST, etc on methods and no PATH whatsoever
+        // for clients, it is also enough to only have @GET, @POST, etc on methods and no PATH whatsoever
         Set<DotName> methodAnnotations = httpAnnotationToMethod.keySet();
         for (DotName methodAnnotation : methodAnnotations) {
             for (AnnotationInstance methodAnnotationInstance : index.getAnnotations(methodAnnotation)) {
@@ -369,7 +360,7 @@ public class ResteasyReactiveScanner {
                 continue;
             }
             possibleSubResources.put(classInfo.name(), classInfo);
-            //we need to also look for all sub classes and interfaces
+            //we need to also look for all subclasses and interfaces
             //they may have type variables that need to be handled
             toScan.addAll(index.getKnownDirectImplementors(classInfo.name()));
             toScan.addAll(index.getKnownDirectSubclasses(classInfo.name()));

--- a/integration-tests/rest-client-reactive-http2/src/main/java/io/quarkus/it/rest/client/http2/AbstractGreetingResource.java
+++ b/integration-tests/rest-client-reactive-http2/src/main/java/io/quarkus/it/rest/client/http2/AbstractGreetingResource.java
@@ -1,0 +1,14 @@
+package io.quarkus.it.rest.client.http2;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+@Path("/greet")
+public abstract class AbstractGreetingResource {
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public abstract String hello();
+}

--- a/integration-tests/rest-client-reactive-http2/src/main/java/io/quarkus/it/rest/client/http2/GreetingResource.java
+++ b/integration-tests/rest-client-reactive-http2/src/main/java/io/quarkus/it/rest/client/http2/GreetingResource.java
@@ -1,0 +1,8 @@
+package io.quarkus.it.rest.client.http2;
+
+public class GreetingResource extends AbstractGreetingResource {
+    @Override
+    public String hello() {
+        return "Hello";
+    }
+}

--- a/integration-tests/rest-client-reactive-http2/src/test/java/io/quarkus/it/rest/client/http2/GreetingResourceTest.java
+++ b/integration-tests/rest-client-reactive-http2/src/test/java/io/quarkus/it/rest/client/http2/GreetingResourceTest.java
@@ -1,0 +1,19 @@
+package io.quarkus.it.rest.client.http2;
+
+import static io.restassured.RestAssured.when;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+class GreetingResourceTest {
+
+    @Test
+    void testGreeting() {
+        when()
+                .get("/greet")
+                .then()
+                .statusCode(200);
+    }
+}

